src/core/EdFi.DataManagementService.Core/ApiService.cs

```diff
@@ -126,10 +145,18 @@ public async Task<IFrontendResponse> Upsert(FrontendRequest frontendRequest)
     /// <summary>
     /// DMS entry point for all API GET by id requests
     /// </summary>
-    public async Task<IFrontendResponse> GetById(FrontendRequest frontendRequest)
+    public async Task<IFrontendResponse> Get(FrontendRequest frontendRequest)
     {
         PipelineContext pipelineContext = new(frontendRequest, RequestMethod.GET);
-        await _getByIdSteps.Value.Run(pipelineContext);
+        string[] segments = frontendRequest.Path.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+        if (segments.Length == 3 && segments[2].Length == 36 && Guid.TryParse(segments[2], out _))
+        {
+            await _getByIdSteps.Value.Run(pipelineContext);
+        }
```

Comment:
What will happen if someone has a request like `GET .../ed-fi/students/not-a-guid`? That should result in a 404, but it looks like it will end up triggering the Get All request below.

Reply:
Is correct it goes by Get All path, however, in one of the pipeline steps (ParsePathMiddleware) it returns a 404

Reply:
`GetById` should stay as-is but not be exposed through the `IApiService` interface. A _new_ `Get` that is exposed should check for a document uuid and either call `GetById` or a new `Query` method. `ParsePathMiddleware` already does this with a regex so reuse that rather than do it yourself with the `segments` stuff.

Meadowlark.js has an example of this new `Get` behavior here:
https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/blob/21eca1ca8eeb0362354f94304821eb73cf9910e7/Meadowlark-js/packages/meadowlark-core/src/handler/FrontendFacade.ts#L174